classes/mail/variables/IssueEmailVariable.php

```diff
@@ -67,4 +76,72 @@ protected function getIssueUrl(): string
+
+        $templateMgr->assign([
+            'issueIdentification' => $this->issue->getIssueIdentification(),
+            'issueTitle' => $this->issue->getLocalizedTitle(),
+            'issueSeries' => $this->issue->getIssueIdentification(['showTitle' => true]),
+            'currentContext' => $this->getContext(),
+            'currentJournal' => $this->getContext(),
+        ]);
+
+        $issueGalleyDao = DAORegistry::getDAO('IssueGalleyDAO'); /** @var IssueGalleyDAO $issueGalleyDao */
+
+        $genreDao = DAORegistry::getDAO('GenreDAO'); /** @var GenreDAO $genreDao */
+        $primaryGenres = $genreDao->getPrimaryByContextId($this->getContext()->getId())->toArray();
+        $primaryGenreIds = array_map(function ($genre) {
+            return $genre->getId();
+        }, $primaryGenres);
+
+        $issueSubmissions = Repo::submission()->getCollector()
+            ->filterByContextIds([$this->issue->getJournalId()])
+            ->filterByIssueIds([$this->issue->getId()])
+            ->filterByStatus([Submission::STATUS_PUBLISHED])
+            ->orderBy(\APP\submission\Collector::ORDERBY_SEQUENCE, \APP\submission\Collector::ORDER_DIR_ASC)
+            ->getMany();
+
+        $sections = Repo::section()->getByIssueId($this->issue->getId());
+        $issueSubmissionsInSection = [];
+
+        foreach ($sections as $section) {
+            $issueSubmissionsInSection[$section->getId()] = [
+                'title' => $section->getHideTitle() ? null : $section->getLocalizedTitle(),
+                'hideAuthor' => $section->getHideAuthor(),
+                'articles' => [],
+            ];
+        }
+
+        foreach ($issueSubmissions as $submission) {
+            if (!$sectionId = $submission->getCurrentPublication()->getData('sectionId')) {
+                continue;
+            }
+            $issueSubmissionsInSection[$sectionId]['articles'][] = $submission;
+        }
+
+        $authorUserGroups = Repo::userGroup()->getCollector()
+            ->filterByRoleIds([\PKP\security\Role::ROLE_ID_AUTHOR])
+            ->filterByContextIds([$this->getContext()->getId()])
+            ->getMany();
+
+        $templateMgr->assign([
+            'issueIdentification' => $this->issue->getIssueIdentification(),
+            'issueTitle' => $this->issue->getLocalizedTitle(),
+            'issueSeries' => $this->issue->getIssueIdentification(['showTitle' => true]),
+            'currentContext' => $this->getContext(),
+            'currentJournal' => $this->getContext(),
+            'issue' => $this->issue,
+            'issueGalleys' => $issueGalleyDao->getByIssueId($this->issue->getId()),
+            'publishedSubmissions' => $issueSubmissionsInSection,
+            'primaryGenreIds' => $primaryGenreIds,
+            'authorUserGroups' => $authorUserGroups,
+            'locale' => Locale::getLocale(),
+        ]);
+
+        return PKPString::stripUnsafeHtml($templateMgr->fetch('frontend/objects/issue_toc.tpl'));
```

Comment:
I'm not sure about this `stripUnsafeHtml` wrapper. Is there specific content in the template that should be stripped? Or is this in lieu of having better control of the formatting?

Reply:
used this to removed the extra lines and white spaces but probably better to not use it as that make the list too compact . I have removed it .